fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/ContainmentIndexImpl.java

```diff
@@ -224,19 +247,32 @@ private NamedParameterJdbcTemplate getNamedParameterJdbcTemplate() {
             // not in a transaction
             children = jdbcTemplate.queryForList(SELECT_CHILDREN, parameterSource, String.class);
         }
+        LOGGER.debug("getChildren for {} in transaction {} found {} children", fedoraId, transactionId,
+                children.size());
         return children.stream();
     }
 
     @Override
-    public Stream<String> getContainedBy(final Transaction tx, final FedoraResource fedoraResource) {
+    public Stream<String> getContains(final Transaction tx, final FedoraResource fedoraResource) {
         final String txId = (tx != null) ? tx.getId() : null;
         return getChildren(fedoraResource.getId(), txId);
     }
 
     @Override
-    public void addContainedBy(final Transaction tx, final FedoraResource parent, final FedoraResource child) {
-        final String txID = tx != null ? tx.getId() : null;
-        addContainedBy(txID, parent.getId(), child.getId());
+    public String getContainedBy(final String txID, final String resourceID) {
+        final MapSqlParameterSource parameterSource = new MapSqlParameterSource();
+        parameterSource.addValue("child", resourceID);
+        final List<String> parentID;
+        if (txID != null) {
+            parameterSource.addValue("transactionId", txID);
+            final String currentResourceQuery = PARENT_EXISTS +
+                    " UNION " + PARENT_EXISTS_ADDITIONS +
```

Comment:
No, the eventual query is all records from the main table for the resource and transaction table with "add" unless they are in the transaction table with "delete"

Reply:
..and it does not matter if the "delete" preceded the "add"?

Reply:
No, because it isn't actually a delete. Its an `EXCEPT`.
> EXCEPT
> Returns any distinct values from the query left of the EXCEPT operator. Those values return as long the right query doesn't return those values as well.

Reply:
Sorry the DELETEs and ADDs make it seem like an update where it is just a complex query. So we need to do the UNION and then the EXCEPT.